src/prefect/serialization/storage.py

```diff
@@ -49,6 +40,8 @@ class Meta:
     flows = fields.Dict(key=fields.Str(), values=fields.Str())
     prefect_version = fields.String(allow_none=False)
     secrets = fields.List(fields.Str(), allow_none=True)
+    labels = fields.List(fields.Str(), allow_none=True)
```

Comment:
I think we should avoid touching the storage serializers in this PR; this adds compatibility issues and I think we should wait until we see how this is used.  Additionally we'll want a unified place for the "source of truth" of what labels are present on a flow, and right now that source of truth is the `Environment`.

Reply:
Fine with me, but depending on whether we let users tag their storages with arbitrary labels, we may run into a scenario where:

```python
storage = Storage(labels=["foo", "bar"])
storage == StorageSchema().load(StorageSchema().dump(storage)) # False
```

Reply:
Yea, that's actually fine - 99% of our serializers don't satisfy the check you wrote actually.  In this case the `labels` on Storage are purely used at registration time (prior to serialization) and never needed again.  As long as the deserialized Storage class can actually _run_ and retrieve the underlying Flow object as users intended it, we're good